File: javascript/monothread-sort/selection-sort.js
function sort(array) {
  for (let i = 0; i < array.length; i++) {
    let index = i;

    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < array[index]) {
        index = j;
      }
    }

    if (index != i) {
      let aux = array[i];

      array[i] = array[index];
      array[index] = aux;
    }
  }
}

module.exports = {
  sort,
};

File: cpp/monothread_sort/selection_sort.cpp
namespace selection_sort
{
  using namespace std;

  void sort(vector<int> *array)
  {
    for (int i = 0; i < array->size(); i++)
    {
      int index = i;

      for (int j = i + 1; j < array->size(); j++)
      {
        if (array->operator[](j) < array->operator[](index))
        {
          index = j;
        }
      }

      if (index != i)
      {
        int aux = array->operator[](i);

        array->operator[](i) = array->operator[](index);
        array->operator[](index) = aux;
      }
    }
  }
}
